### PR TITLE
ml_smoothing cleaned up

### PR DESCRIPTION
**Pull request by harvey2phase**

### CODE REVIEW

File: cycling/management/commands/compile_dataset.py [hunk 1]
```diff
@@ -298,11 +298,11 @@ def initial_processing(my_cell_ids, fit_args, flags):
 
         cycle_span = max_cycle - min_cycle
 
-        delta_cycle = cycle_span / float(fit_args[Key.REF_CYC])
+        delta_cycle = cycle_span / float(fit_args[Key.REF_CYC_N])
```
Comment (Samuel-Buteau) [inline cycling/management/commands/compile_dataset.py:301]: use numpy.linspace instead

File: machine_learning/management/commands/ml_smoothing.py [hunk 2]
```diff
@@ -36,89 +22,97 @@
...
 NEIGH_RATE = 2
 NEIGH_CELL_ID = 3
-NEIGH_ABSOLUTE_CYCLE = 4
+NEIGH_ABSOLUTE_CYC = 4
 NEIGH_VALID_CYC = 5
 NEIGH_SIGN_GRID = 6
 NEIGH_VOLTAGE_GRID = 7
 NEIGH_CURRENT_GRID = 8
-NEIGH_TEMPERATURE_GRID = 9
+NEIGH_TMP_GRID = 9
```
Comment (Samuel-Buteau) [inline machine_learning/management/commands/ml_smoothing.py:30]: Really? 
this is a bad abreviation
Maybe NEIGH_T_GRID

File: machine_learning/management/commands/ml_smoothing.py [hunk 3]
```diff
@@ -511,11 +480,32 @@ def initial_processing(
...
-        Key.DATASET: my_data,
+        Key.DATASET: dataset,
     }
 
 
-def train_and_evaluate(init_returns, cell_ids, fit_args):
+def to_sorted_array(unsorted) -> np.array:
+    """ Turn a list or a view object (of a dictionary) to a sorted array
+
+    Args:
```
Comment (Samuel-Buteau) [inline machine_learning/management/commands/ml_smoothing.py:490]: also removes duplicates

File: cycling/management/commands/compile_dataset.py (hunk 1 above)
Comment (harvey2phase) [reply cycling/management/commands/compile_dataset.py:301]: Like replace the block with
```python
numpy.linspace(min_cycle + delta_cycle, max_cycle, fit_args[Key.REF_CYC_N])
```
?

File: machine_learning/management/commands/ml_smoothing.py (hunk 2 above)
Comment (harvey2phase) [reply machine_learning/management/commands/ml_smoothing.py:30]: I thought `TMP` would be less ambiguous if it was in lowercase, or we could use `tmp` in lowercase and `T` in uppercase?

File: cycling/management/commands/compile_dataset.py (hunk 1 above)
Comment (harvey2phase) [reply cycling/management/commands/compile_dataset.py:301]: Like replace the block with
```python
numpy.linspace(min_cycle + delta_cycle, max_cycle, fit_args[Key.REF_CYC_N])
```
?

**Pull request merged**